### PR TITLE
Refactor `MarkReadImplTest`

### PR DESCRIPTION
**Pull request by samiuelson**
https://stream-io.atlassian.net/browse/CAS-598
### Description

Refactor `MarkReadImplTest`
* Decouple from `BaseDomainTest`
* Make it unit test

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/usecase/MarkReadImplTest.kt [hunk 1]
```diff
@@ -1,31 +1,94 @@
...
-        Truth.assertThat(channelControllerImpl.toChannel().unreadCount).isEqualTo(0)
+    fun `Given invalid cid Should throw exception`() {
+        assertThrows(IllegalArgumentException::class.java) {
+            sut.invoke("no-colon")
+        }
+    }
+
+    @Test
+    fun `Given valid cid Should invoke markRead() on ChannelController`() = runBlockingTest {
+        val channelController = mock<ChannelControllerImpl>()
```
Comment (ogkuzmin) [inline stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/usecase/MarkReadImplTest.kt:63]: It's always good to split test setup, test action and test assert by empty lines. It let you distinguish these blocks briefly

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/usecase/MarkReadImplTest.kt [hunk 2]
```diff
@@ -1,31 +1,94 @@
...
+        val cid = randomCID()
+        whenever(chatDomain.channel(cid)) doReturn channelController
+        sut.invoke(cid).execute()
+        verify(channelController).markRead()
+    }
+
+    @Test
+    fun `Given valid cid and successful ChannelController result Should return call client markRead() and return Result(true)`() = runBlockingTest {
+        val channelController = mock<ChannelControllerImpl> {
```
Comment (ogkuzmin) [inline stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/usecase/MarkReadImplTest.kt:72]: Too much noise for this test. The Fixture pattern lets you remove those noise and hide it under the hood, it makes test clean and readable. Maybe it is worth using it here.

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/usecase/MarkReadImplTest.kt [hunk 3]
```diff
@@ -1,31 +1,94 @@
...
+        whenever(channelController.channelType) doReturn channelType
+        whenever(channelController.channelId) doReturn channelId
+        val markReadCall = mock<Call<Unit>>()
+        whenever(client.markRead(any(), any())) doReturn markReadCall
+        val markReadCallResult = mock<Result<Unit>>()
+        whenever(markReadCall.execute()) doReturn markReadCallResult
+
+        val result = sut.invoke(cid).execute()
+
+        verify(channelController).markRead()
```
Comment (ogkuzmin) [inline stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/usecase/MarkReadImplTest.kt:89]: IMO in this test, it's better to split it for two. Usually, they are two different kinds of tests.
1) `State tests` - when you check the state of some objects (checking fields to be equal to some values)
2) `Behavioral tests` - when you check that something happened (mockito's `verify`)

Comment (samiuelson) [reply stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/usecase/MarkReadImplTest.kt]: done

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/usecase/MarkReadImplTest.kt (hunk 2 above)
Comment (samiuelson) [reply stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/usecase/MarkReadImplTest.kt]: I made it cleaner.

**Review by ogkuzmin — APPROVED** [review]
LGTM

**Pull request merged**